### ISSUE TITLE
Path and module changes (RFC 2126)

### ISSUE THREAD

**Issue by matthewjasper** [rust-lang-nursery/reference#291]
See rust-lang/rust#44660.

- [ ] Paths starting with `crate`
- [ ] Not needing `extern crate`
- [ ] `crate` visibility modifier
- [ ] Not using `mod.rs`
- [ ] Any edition lints
- [ ] Update existing examples

**Comment by ehuss** [comment]
I have a branch for the `extern crate` changes, but it is blocked waiting on #415 (I either need to link to or write something that explains the extern prelude).  `crate` visibility is not stable (rust-lang/rust#53120) so I don't think that's relevant here.  I opened PR #450 for `mod.rs` changes.

I'm not sure what is meant by "edition lints".  My `extern crate` changes note that it is unidiomatic in 2018, but I'm not sure what other lints are relevant.  Also, I was under the impression lints aren't generally documented here and deferred to the rustc user guide.  Are there specific `rust_2018_idioms` you want to call out?

**Comment by ehuss** [comment]
Regarding updating the examples, AFAIK the mdbook playpen links don't support changing the edition, so those links will always break.  I can't find this being tracked or mentioned anywhere.  I imagine they aren't used very often, but I think it should be tracked somewhere (should I file an mdbook issue?).
